src/core/midiDispatcher.cpp

```diff
@@ -25,403 +25,240 @@
+	// 'n' sends messages to null ;)
+	// Note that only the first character is read from the first part of 
+	// an address, so one might as well put "port" or "channel" in there.
+	//
+	// The second part is an actual address
+	// Should any following parts occur, those are for future use.
+	
+	// Parse the address into 'parsed_r' vector
+	std::vector<std::string> parsed_r;
+	std::stringstream ss(r);
+	while (ss.good()){
+		std::string str;
+		getline(ss, str, ';');
+		parsed_r.push_back(str);
+	}
 
-/* -------------------------------------------------------------------------- */
+	// Interpret the address and call appropriate midiReceive function
+	switch(parsed_r[0][0]){
+		case 'p':
+
+			midiPorts::midiReceive(mm, parsed_r[1]);		
+			break;
+
+/*
+ * Here messages will be send to channels directly
+ * Not implemented at this stage, midiController does that the old way
+ 
+		case 'c':
+			ID _id;
+			_id = std::stoi(parsed_r[1]);
+				//TODO
+			break;
+*/
+
+		case 'm':
+
+			if (parsed_r[1]=="midiController")
+				midiController::midiReceive(mm);
+			if (parsed_r[1]=="midiDevice")
+				midiDevice::midiReceive(mm);
+			
+	}
 
 
-void clearMasterLearn(int param, std::function<void()> f)
-{
-	learnMaster_(MidiEvent(), param, f); // Empty event (0x0)
 }
 
+} //------------------------------- {anonymous} -------------------------------
 
-void clearChannelLearn(int param, ID channelId, std::function<void()> f)
-{
-	learnChannel_(MidiEvent(), param, channelId, f); // Empty event (0x0)
+void reg(std::string s, MidiMsgFilter& mmf, std::string& r, bool wl){
+	dispatchTable.push_back(DispatchTableItem(s, mmf, r, wl));
 }
 
+//   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -
 
-#ifdef WITH_VST
-
-void clearPluginLearn (int paramIndex, ID pluginId, std::function<void()> f)
-{
-	learnPlugin_(MidiEvent(), paramIndex, pluginId, f); // Empty event (0x0)
+void regEx(std::string s, MidiMsgFilter& mmf, std::string& r, bool wl){
+	dispatchTableEx.push_back(DispatchTableItem(s, mmf, r, wl));
 }
 
-#endif
-
-
-/* -------------------------------------------------------------------------- */
+//   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -
 
+void dispatch(MidiMsg& mm){
 
-void dispatch(int byte1, int byte2, int byte3)
-{
-	/* Here we want to catch two things: a) note on/note off from a keyboard and 
-	b) knob/wheel/slider movements from a controller. 
-	We must also fix the velocity zero issue for those devices that sends NOTE
-	OFF events as NOTE ON + velocity zero. Let's make it a real NOTE OFF event. */
-
-	MidiEvent midiEvent(byte1, byte2, byte3);
-	midiEvent.fixVelocityZero();
-
-	u::log::print("[midiDispatcher] MIDI received - 0x%X (chan %d)\n", midiEvent.getRaw(), 
-		midiEvent.getChannel());
+	// Consult dispatchTableEx first
+	// If a match is found, send message and ignore all the rest
+	for (DispatchTableItem& dti : dispatchTableEx){
+		if (dti.check(mm)){
```

Comment:
```suggestion
	for (DispatchTableItem& dti : dispatchTableEx){
		if (dti.check(mm)) {
```
looks more like the local coding style